### PR TITLE
EZP-28495: As Editor I want UI for external links management

### PR DESCRIPTION
**Pull request by adamwojs**
| Question      | Answer
| ------------- | ---
| Tickets       | https://jira.ez.no/browse/EZP-28495
| Bug fix?      | no 
| New feature?  | yes
| BC breaks?    | no
| Tests pass?   | yes
| Doc needed?   | yes
| License       | [GPL-2.0](https://github.com/ezsystems/ezplatform-admin-ui/blob/master/LICENSE)

This PR contains UI implementation for link management.

Requires the following PRs to be merged first:
* https://github.com/ezsystems/repository-forms/pull/185 
* https://github.com/ezsystems/ezpublish-kernel/pull/2183

### Screenshots

screenshot-2017-12-13 link manager
screenshot-2017-12-13 link http calendar google com calendar render
screenshot-2017-12-13 ez platform

#### Checklist:
- [x] Coding standards (`$ composer fix-cs`)
- [x] Ready for Code Review

### CODE REVIEW

File: src/bundle/Controller/LinkManagerController.php [hunk 1]
```diff
@@ -0,0 +1,217 @@
...
+     * @param Request $request
+     * @param int $urlId ID of URL
+     *
+     * @return Response
+     */
+    public function editAction(Request $request, $urlId)
+    {
+        $url = $this->urlService->loadById($urlId);
+
+        $form = $this->createEditForm($url);
```
Comment (mikadamczyk) [inline src/bundle/Controller/LinkManagerController.php]: This should be moved to facoryForm class

File: src/bundle/Controller/LinkManagerController.php [hunk 2]
```diff
@@ -0,0 +1,217 @@
...
+        $form->handleRequest($request);
+        if ($form->isValid()) {
```
Comment (mikadamczyk) [inline src/bundle/Controller/LinkManagerController.php]: if ($form->isSubmitted() && $form->isValid()) {

File: src/bundle/Controller/LinkManagerController.php [hunk 3]
```diff
@@ -0,0 +1,217 @@
...
+    }
+
+    /**
+     * Creates URL list form.
+     *
+     * @param URLListData|null $data
+     *
+     * @return FormInterface
+     */
+    protected function createListForm(URLListData $data = null)
```
Comment (mikadamczyk) [inline src/bundle/Controller/LinkManagerController.php]: I'm not sure that this method is necessary

File: src/bundle/Resources/views/link_manager/edit.html.twig [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+                        iconName: 'link'
+                    } %}
+                </div>
+            </div>
+
+            <div class="container px-0 pb-4 mt-5">
+                {{ form_errors(form) }}
+
+                <section class="container mt-4">
+                    <div class="card ez-card__form">
```
Comment (mikadamczyk) [inline src/bundle/Resources/views/link_manager/edit.html.twig]: Try ez-card, without "__form"

File: src/bundle/Resources/views/link_manager/list.html.twig [hunk 5]
```diff
@@ -0,0 +1,126 @@
...
+
+{% block javascripts %}
+    {%  javascripts
+        'bundles/ezplatformadminui/js/scripts/admin.linkmanager.list.js'
+    %}
+        <script type="text/javascript" src="{{ asset_url }}"></script>
+    {% endjavascripts %}
+{% endblock %}
+
+
```
Comment (mikadamczyk) [inline src/bundle/Resources/views/link_manager/list.html.twig]: Unnecessary empty line

File: src/bundle/Resources/views/link_manager/list.html.twig [hunk 6]
```diff
@@ -0,0 +1,126 @@
...
+{% block title %}{{ 'url.list'|trans }}{% endblock %}
```
Comment (mikadamczyk) [inline src/bundle/Resources/views/link_manager/list.html.twig]: Maybe title block fits better to the top of the file?

File: src/bundle/Resources/views/link_manager/view.html.twig [hunk 7]
```diff
@@ -0,0 +1,125 @@
...
+                    <td>{{ 'url.label.modified'|trans }}</td>
+                    <td>{{ url.modified | date('M d, Y h:i:sA') }}</td>
+                </tr>
+            </tbody>
+        </table>
+    </section>
+
+    <section class="container mt-5">
+        <div class="ez-table-header round">
+            <h2 class="ez-table-header--label">
```
Comment (mikadamczyk) [inline src/bundle/Resources/views/link_manager/view.html.twig]: What is this '--label' for?

**Comment by adamwojs** [comment]
@mikadamczyk I've attached screenshots as you requested 😉

**Comment by adamwojs** [comment]
@mikadamczyk PR updated according to @mikadamczyk suggestions

File: src/bundle/Resources/public/js/scripts/admin.linkmanager.list.js [hunk 8]
```diff
@@ -0,0 +1,7 @@
+(function(doc) {
+    const statusField = doc.getElementById('search_data_status');
+
+    statusField.addEventListener('change', function() {
+        this.form.submit();
+    });
+}) (document);
```
Comment (sunpietro) [inline src/bundle/Resources/public/js/scripts/admin.linkmanager.list.js]: CS: no extra space needed.

File: src/bundle/Resources/public/js/scripts/admin.linkmanager.list.js [hunk 9]
```diff
@@ -0,0 +1,7 @@
+(function(doc) {
+    const statusField = doc.getElementById('search_data_status');
```
Comment (sunpietro) [inline src/bundle/Resources/public/js/scripts/admin.linkmanager.list.js]: What is the HTML structure of element with id `search_data_status`?

File: src/bundle/Resources/views/link_manager/list.html.twig [hunk 10]
```diff
@@ -0,0 +1,124 @@
...
+                </div>
+            </div>
+
+        {{ form_widget(form.page, { attr: { value: '1' }}) }}
+        {{ form_end(form) }}
+    </section>
+
+    <section class="container mt-4">
+        <div class="ez-table-header round">
+            <h5 class="ez-table-header">
```
Comment (sunpietro) [inline src/bundle/Resources/views/link_manager/list.html.twig]: It's not `h5` element anymore. It's a `div` now.

File: src/bundle/Resources/views/link_manager/view.html.twig [hunk 11]
```diff
@@ -0,0 +1,125 @@
...
+                    <td>{{ 'url.label.modified'|trans }}</td>
+                    <td>{{ url.modified | date('M d, Y h:i:sA') }}</td>
+                </tr>
+            </tbody>
+        </table>
+    </section>
+
+    <section class="container mt-5">
+        <div class="ez-table-header round">
+            <h5 class="ez-table-header">
```
Comment (dew326) [inline src/bundle/Resources/views/link_manager/view.html.twig]: This should be: `<div class="ez-table-header__headline">Text</div>`

File: src/bundle/Resources/views/link_manager/view.html.twig [hunk 12]
```diff
@@ -0,0 +1,125 @@
...
+                {{ 'url.tab.usages'|trans({'%count%': usages.nbResults })|desc('Content item(s) using this URL') }}
+            </h5>
+        </div>
+
+        <table class="table">
+            <colgroup>
```
Comment (sunpietro) [inline src/bundle/Resources/views/link_manager/view.html.twig]: Do you need it?

File: src/bundle/Resources/views/link_manager/list.html.twig [hunk 13]
```diff
@@ -0,0 +1,124 @@
...
+                    <td>{{ url.modified | date('M d, Y h:i:sA') }}</td>
+                    <td class="text-right">
+                        {% if can_edit %}
+                            <a href="{{ edit_url }}" class="btn btn-icon">
+                                <svg class="ez-icon ez-icon-edit">
+                                    <use xlink:href="{{ asset('bundles/ezplatformadminui/img/ez-icons.svg') }}#edit"></use>
+                                </svg>
+                            </a>
+                        {% else %}
+                            <span class="btn btn-icon">
```
Comment (dew326) [inline src/bundle/Resources/views/link_manager/list.html.twig]: What is the point to show edit icon when user cannot edit?

File: src/bundle/Resources/views/link_manager/list.html.twig [hunk 14]
```diff
@@ -0,0 +1,124 @@
...
+                    </div>
+                </div>
+            </div>
+
+        {{ form_widget(form.page, { attr: { value: '1' }}) }}
+        {{ form_end(form) }}
+    </section>
+
+    <section class="container mt-4">
+        <div class="ez-table-header round">
```
Comment (mikadamczyk) [inline src/bundle/Resources/views/link_manager/list.html.twig]: I think .round is not needed anymore

File: src/bundle/Resources/views/link_manager/view.html.twig [hunk 15]
```diff
@@ -0,0 +1,125 @@
...
+{%- endblock -%}
+
+{%- block pageTitle -%}
+    {% include '@EzPlatformAdminUi/parts/page_title.html.twig' with {
+        title: url.url|truncate(60),
+        iconName: 'link'
+    } %}
+{%- endblock -%}
+
+{% block bodyClass %}ez-info-view{% endblock %}
```
Comment (dew326) [inline src/bundle/Resources/views/link_manager/view.html.twig]: This class is for `System Information`, please use `ez-link-manager-view` or something similar.

File: src/bundle/Resources/public/js/scripts/admin.linkmanager.list.js (hunk 9 above)
Comment (adamwojs) [reply src/bundle/Resources/public/js/scripts/admin.linkmanager.list.js]: `#search_data_status`  is `<select>` (link status filter)

**Comment by adamwojs** [comment]
PR updated according to @sunpietro, @dew326, @mikadamczyk suggestions.

**Pull request merged**